docs/flinkDev/building.md

```diff
@@ -97,35 +97,49 @@ mvn clean install
 
 ## Hadoop Versions
 
-{% info %} Most users do not need to do this manually. The [download page]({{ site.download_url }}) contains binary packages for common Hadoop versions.
+Flink has optional dependencies to HDFS and YARN which are both dependencies from [Apache Hadoop](http://hadoop.apache.org). There exist many different versions of Hadoop (from both the upstream project and the different Hadoop distributions). If you are using a wrong combination of versions, exceptions can occur.
 
-Flink has dependencies to HDFS and YARN which are both dependencies from [Apache Hadoop](http://hadoop.apache.org). There exist many different versions of Hadoop (from both the upstream project and the different Hadoop distributions). If you are using a wrong combination of versions, exceptions can occur.
+Flink can be built against any Hadoop version >= 2.4.0, but depending on the version it may be a 1 or 2 step process.
 
-Hadoop is only supported from version 2.4.0 upwards.
-You can also specify a specific Hadoop version to build against:
+### Pre-bundled versions
+
+If you want to build against Hadoop 2.4.1, 2.6.5, 2.7.5 or 2.8.3, then it is sufficient to run:
+
+{% highlight bash %}
+mvn clean install -DskipTests -Dhadoop.version=2.6.5
+{% endhighlight %}
+
+To further package a shaded pre-packaged hadoop jar into the distribution, additionally activate the `include-hadoop` profile`:
 
 {% highlight bash %}
-mvn clean install -DskipTests -Dhadoop.version=2.6.1
+mvn clean install -DskipTests -Dhadoop.version=2.6.5 -Pinclude-hadoop
 {% endhighlight %}
 
-### Packaging Hadoop into the Flink distribution
+### Custom / Vendor-specific versions
+
+If you want to build against Hadoop version that is *NOT* 2.4.1, 2.6.5, 2.7.5 or 2.8.3,
+then it is first necessary to build [flink-shaded](https://github.com/apache/flink-shaded) against this version.
+You can find the source for this project in the `Additional Components` section of the [downloads page](http://flink.apache.org/downloads.html#additional-components).
```

Comment:
Suggestions:
> against **a** Hadoop version ...
> then it is necessary to build [flink-shaded]... **first** against that version.

Use `{{ site.download_url }}` instead of explicit `http://flink.apache.org/downloads.html` url.

Also, personally, I'd change the url link to be anchored with `Additional Components` text:
> You can **also** download the source code of this project in the [Additional Components]({{ site.download_url }}#additional-components) section of the downloads page.